lib/urldata.h

```diff
@@ -201,6 +201,8 @@
 /* Download buffer size, keep it fairly big for speed reasons */
 #undef BUFSIZE
 #define BUFSIZE CURL_MAX_WRITE_SIZE
+#define MAX_BUFSIZE 524288
```

Comment:
Each SFTP packet needs to get requested individually, and sent back. That means that to fill up 512KB it needs 55 milliseconds (if we rule out a lot of other complications). It certainly can't do it any faster at least. (Because of how curl always drains the entire buffer first before it asks for more data.)

~~~
MAX = (512*1024) / 0.055 = 9532509 bytes/sec
~~~
That seems to match your numbers for the 512KB buffer case, but since you can't go faster with a larger buffer, it might indicate that limit is then not in the client SFTP code and buffer anymore.

But if we say talk with a more remote server at for example 200ms RTT, the max speed with 512KB of data buffer is:
~~~
MAX = (512 * 1024)/0.200 = 2621440 bytes/sec
~~~
While a doubled buffer size quite should be able to double it:
~~~
MAX = (2* 512 * 1024)/0.200 = 5242880/sec
~~~